packages/platform/src/HttpApiEndpoint.ts

```diff
@@ -728,10 +759,18 @@ const Proto = {
       )
     })
   },
-  setPayload(this: HttpApiEndpoint.AnyWithProps, schema: Schema.Schema.Any) {
+  setPayload(
+    this: HttpApiEndpoint.AnyWithProps,
+    payload: Schema.Schema.Any | {
+      readonly stream: true
+      readonly contentType: string
+      readonly contentLength?: number | undefined
+      readonly etag?: string | undefined
+    }
+  ) {
     return makeProto({
       ...this,
-      payloadSchema: Option.some(schema)
+      payloadSchema: Option.some(payload)
     })
   },
```

Comment:
Thoughts on adding a `stream` option to setPayload? Is this something you think could make it in with more refinement?